File: nodes/automl-predictor/automl-predictor.py
import json
import pandas
import os
import sys
import io
sys.path.append(os.path.dirname(os.path.realpath(__file__)) + '/../../utils')
from sklw import NumpyEncoder

#read configurations
config = json.loads(input())
model=None
def load():
	try:
		from sklw import SKLW
		return SKLW(path=config['path'])
	except:
		return None

while True:
	data=input()
	if "orient" in data:
		new_config = json.loads(data)
		if new_config.get('modelPath') and new_config.get('modelName'):
			config['path']=os.path.join(new_config['modelPath'], new_config['modelName'])
		if new_config.get('orient'):
			config['orient']=new_config['orient']
		model = load()
		print(json.dumps({"state":"parameters applied","config":config}))
		continue

	#read request
	features = pandas.read_json(io.StringIO(data), orient=config['orient'])

	if model is None:
		model = load()
	if model is None:
		raise('Cannot find model.')
	model.update()
	
	print(json.dumps({"predict":model.predict(features).data.tolist()},cls=NumpyEncoder))

File: nodes/automl-predictor/automl-predictor.js
module.exports = function(RED){
  function autoMLpredictorNode(config){
    const path = require('path')
    const utils = require('../../utils/utils')

    var node = this;
    
    //set configurations
    node.file = __dirname + '/automl-predictor.py'
    node.topic = 'predicted'
    node.config = {
      path: path.join(config.modelPath, config.modelName),
      orient: config.orient
    }

    utils.run(RED, node, config)
  }
  RED.nodes.registerType("automl-predictor", autoMLpredictorNode)
}
